### PR TITLE
ref(access logs): Add rate limit group

### PR DESCRIPTION
**Pull request by ceorourke**
Add the `RateLimitConfig` group to the API access logs. Right now the only value is "default", but as the project continues endpoints may define their own groups. To accomplish this I moved some things around - `get_rate_limit_key` and `get_rate_limit_value` were each calling `get_rate_limit_config` , so I instead pass the value of `get_rate_limit_config` to both functions and only calculate it once, as well as put it on the request object so that it can be accessed in the API access log middleware. 

This is a redo of https://github.com/getsentry/sentry/pull/33476 - I had initially panic reverted it in https://github.com/getsentry/sentry/pull/33516 because it was causing [SENTRY-TP9](https://sentry.io/organizations/sentry/issues/3184564140/?project=1), but have since dug into it and realized I didn't actually break _all_ access logs, just ones that didn't have a view. This PR also approaches the change in a different and hopefully better way.

### CODE REVIEW

File: src/sentry/middleware/ratelimit.py [hunk 1]
```diff
@@ -40,13 +41,21 @@ def __call__(self, request: Request) -> Response:
...
             try:
                 # TODO: put these fields into their own object
                 request.will_be_rate_limited = False
                 request.rate_limit_category = None
                 request.rate_limit_uid = uuid.uuid4().hex
-                request.rate_limit_key = get_rate_limit_key(view_func, request)
+                view_class = getattr(view_func, "view_class", None)
+                if view_class:
+                    rate_limit_config = get_rate_limit_config(view_class)
+                request.rate_limit_group = (
```
Comment (volokluev) [inline src/sentry/middleware/ratelimit.py:55]: I see you're putting the rate limit group on the request. Remember how we talked about making state that you attach to the global context purposeful, contained, and immutable. Does attaching the group to the request directly serve those heuristics? Also, if the config changes somehow or we wanted to attach different parts of it to the request, how would this code have to change?

**Review by volokluev — COMMENTED** [review]
This looks good to me, but I should not be the one to approve this PR, make sure someone else from ecosystem looks over it (preferably not Aniket) and asks you as many questions as they can

File: src/sentry/ratelimits/utils.py [hunk 2]
```diff
@@ -120,17 +124,21 @@ def get_rate_limit_config(endpoint: Type[object]) -> RateLimitConfig | None:
...
 
 def get_rate_limit_value(
-    http_method: str, endpoint: Type[object], category: RateLimitCategory
+    http_method: str,
+    endpoint: Type[object],
+    category: RateLimitCategory,
+    rate_limit_config: RateLimitConfig | None,
 ) -> RateLimit | None:
     """Read the rate limit from the view function to be used for the rate limit check."""
     # types are hashable in python, the type checker disagrees though
```
Comment (mgaeta) [inline src/sentry/ratelimits/utils.py:133]: This comment was for the deleted line.

Comment (mgaeta) [reply src/sentry/ratelimits/utils.py:133]: Also now the `endpoint` parameter isn't being read so we can take it out.

File: src/sentry/ratelimits/utils.py [hunk 3]
```diff
@@ -120,17 +124,21 @@ def get_rate_limit_config(endpoint: Type[object]) -> RateLimitConfig | None:
...
     if not rate_limit_config:
         return None
     return rate_limit_config.get_rate_limit(http_method, category)
 
 
-def above_rate_limit_check(key: str, rate_limit: RateLimit, request_uid: str) -> RateLimitMeta:
+def above_rate_limit_check(
+    key: str, rate_limit: RateLimit, request_uid: str, group: str
+) -> RateLimitMeta:
     # TODO: This is not as performant as it could be. The roundtrip betwwen the server and redis
```
Comment (mgaeta) [inline src/sentry/ratelimits/utils.py:142]: `betwwen` -> `between`

**Review by mgaeta — APPROVED** [review]

**Pull request merged**